packages/jsdoc-plugins/lib/validator/doclet-validator.js

```diff
@@ -316,6 +312,9 @@ class DocletValidator {
 	 * @returns {Boolean}
 	 */
 	_isCorrectType( type ) {
+		// JSDoc converts `Type.<Function>` to `Type.<function()>` for some reason...
+		type = type.replace( 'function()', 'function' );
```

Comment:
I'm not sure about this line. I noticed that in some places we had a strange `function()` param... E.g:

![image](https://user-images.githubusercontent.com/156149/27325275-2bc2c15a-55a8-11e7-9048-11b0163ae917.png)

The documentation of this method looks like this:

```
	/**
	 * Loads a set of plugins and adds them to the collection.
	 *
	 * @param {Array.<Function|String>} plugins An array of {@link module:core/plugin~PluginInterface plugin constructors}
	 * or {@link module:core/plugin~PluginInterface.pluginName plugin names}. The second option (names) work only if
	 * `availablePlugins` were passed to the {@link #constructor}.
	 * @param {Array.<String|Function>} [removePlugins] Names of plugins or plugin constructors
	 * which should not be loaded (despite being specified in the `plugins` array).
	 * @returns {Promise} A promise which gets resolved once all plugins are loaded and available into the
	 * collection.
	 * @returns {Promise.<Array.<module:core/plugin~PluginInterface>>} returns.loadedPlugins The array of loaded plugins.
	 */
```

And I'd like there the `Function` type, not `function()` nor `function`.